mlflow/projects/_project_spec.py

```diff
@@ -102,14 +104,12 @@ def compute_parameters(self, user_parameters, storage_dir):
         final_params = {}
         extra_params = {}
 
-        for name, param_obj in self.parameters.items():
-            if name in user_parameters:
-                final_params[name] = param_obj.compute_value(user_parameters[name], storage_dir)
-            else:
-                final_params[name] = self.parameters[name].default
-        for name in user_parameters:
-            if name not in final_params:
-                extra_params[name] = user_parameters[name]
+        for key, param_obj in self.parameters.items():
+            value = user_parameters[key] if key in user_parameters else self.parameters[key].default
+            final_params[key] = param_obj.compute_value(value, storage_dir)
```

Comment:
Does this mean that we'll log a local path instead of a URI for things that we downloaded of type `path`? That would be a problem -- we want to essentially log the same command the user ran. Do we test for that?

Reply:
No. This method is reused for parameter resolution during training and logging. Logging does not use local path (since `storage_dir = None`).

Added a comment at the top of this method

Reply:
OK, that makes sense. I was confused by it.

Reply:
It may still be good to add a test with path if we don't have one already though, in case this breaks later!

Reply:
Will do.